spec/controllers/web/dashboard/skills_controller_spec.rb

```diff
@@ -0,0 +1,318 @@
+          put :update, params: { id: skill.id, skill: new_skill_params }
+          skill.reload
+          expect(skill.title).to eq new_skill_params[:title]
+        end
+
+        it 'redirects to dashboard skills' do
+          put :update, params: { id: skill.id, skill: new_skill_params }
+          expect(response).to redirect_to dashboard_skills_url
+        end
+      end
+
+      context 'user has role developer or author' do
+        let(:user) { create(:user, %i[developer author].sample, :active) }
+
+        it 'redirects to dashboard root' do
+          put :update, params: { id: skill.id, skill: new_skill_params }
+          expect(response).to redirect_to dashboard_root_url
+        end
+      end
+
+      context 'not passed validation' do
+        let(:user) { create(:user, :staff, :active) }
+        let!(:new_skill_params) do
+          {
+            title: nil
+          }
+        end
+
+        it 'render edit' do
+          put :update, params: { id: skill.id, skill: new_skill_params }
+          expect(response).to render_template :edit
+        end
+      end
+    end
+  end
+
+  describe 'GET #activate' do
+    let(:skill) { create(:skill, :disabled) }
+
+    context 'not signed in' do
+      let(:user) { create(:user, :staff, :active) }
+      before { put :activate, params: { id: skill.id } }
+
+      it 'redirects to root landing' do
+        expect(response).to redirect_to root_landing_url
+      end
+    end
+
+    context 'signed in' do
+      before { login_user(user) }
+
+      context 'user has role staff or mentor' do
+        let(:user) { create(:user, %i[staff mentor].sample, :active) }
+
+        it 'the skill became active' do
+          put :activate, params: { id: skill.id }
+          skill.reload
+          expect(skill.state).to eq 'active'
+        end
+
+        it 'redirects to dashboard skills' do
+          put :activate, params: { id: skill.id }
+          expect(response).to redirect_to dashboard_skills_url
+        end
+      end
+
+      context 'user has role developer or author' do
+        let(:user) { create(:user, %i[developer author].sample, :active) }
+
+        it 'redirects to dashboard root' do
+          put :activate, params: { id: skill.id }
+          expect(response).to redirect_to dashboard_root_url
+        end
+      end
+    end
+  end
+
+  describe 'GET #deactivate' do
+    let(:skill) { create(:skill, :active) }
+
+    context 'not signed in' do
+      let(:user) { create(:user, :staff, :active) }
+      before { put :deactivate, params: { id: skill.id } }
+
+      it 'redirects to root landing' do
+        expect(response).to redirect_to root_landing_url
+      end
+    end
+
+    context 'signed in' do
+      before { login_user(user) }
+
+      context 'user has role staff or mentor' do
+        let(:user) { create(:user, %i[staff mentor].sample, :active) }
+
+        it 'the skill became disabled' do
+          put :deactivate, params: { id: skill.id }
```

Comment:
expect { put ... }.to change { skill.reload.state }.from('active').to('disabled')